p2p/src/connection/p2p_connection_state.rs

```diff
@@ -20,6 +20,7 @@ impl P2pConnectionState {
             time: Timestamp::ZERO,
             opts: opts.clone(),
             rpc_id: None,
+            on_success: None,
```

Comment:
same here, does it make sense for this to be `None` I see that in the only call that exists to this constructor, the callback is immediately assigned after

Reply:
I think that this is just a helper function to create "empty" state if it doesn't exist, if we created actual state we would have to create that state twice, once if it doesn't exist to create it and then once to update it